libs/ledger-live-common/src/families/elrond/hw-app-elrond/index.ts

```diff
@@ -180,4 +182,60 @@ export default class Elrond {
     const signature = response.slice(1, response.length - 2).toString("hex");
     return signature;
   }
+
+  serializeESDTInfo(
+    ticker: string,
+    id: string,
+    decimals: number,
+    chainId: string,
+    signature: string
+  ): Buffer {
+    const tickerLengthBuffer = Buffer.from([ticker.length]);
+    const tickerBuffer = Buffer.from(ticker);
+    const idLengthBuffer = Buffer.from([id.length]);
+    const idBuffer = Buffer.from(id);
+    const decimalsBuffer = Buffer.from([decimals]);
+    const chainIdLengthBuffer = Buffer.from([chainId.length]);
+    const chainIdBuffer = Buffer.from(chainId);
+    const signatureBuffer = Buffer.from(signature, "hex");
+    const infoBuffer = [
+      tickerLengthBuffer,
+      tickerBuffer,
+      idLengthBuffer,
+      idBuffer,
+      decimalsBuffer,
+      chainIdLengthBuffer,
+      chainIdBuffer,
+      signatureBuffer,
+    ];
+    return Buffer.concat(infoBuffer);
+  }
+
+  async provideESDTInfo(
```

Comment:
Not sure what happens with the frontends in this case.

I think a token will not appear in the UI if it isn't in [this list](https://github.com/LedgerHQ/ledger-live/blob/develop/libs/ledgerjs/packages/cryptoassets/src/tokens.ts#L17), right?

Reply:
My bad you're right :slightly_smiling_face: we can't remove completely the file, which is indeed required to display the tokens in the UI, but since tokens aren't hardcoded in the Nano app (unlike other chains) the good news is that we can update the list to add new supported tokens whenever needed.
Thanks for clarifying.